e2e_playwright/st_dataframe_selections_test.py

```diff
@@ -0,0 +1,310 @@
+    _click_on_column_selector(canvas, 4)
+    app.keyboard.up(_command_key)
+    wait_for_app_run(app)
+
+    expected = "Dataframe multi-column selection: {'select': {'rows': [], 'columns': ['col_1', 'col_3', 'col_4']}}"
+    selection_text = app.get_by_test_id("stMarkdownContainer").filter(has_text=expected)
+    expect(selection_text).to_have_count(1)
+
+
+def _select_some_rows_and_columns(app: Page, canvas: Locator):
+    _click_on_row_selector(canvas, 1)
+    _click_on_column_selector(canvas, 1)
+    app.keyboard.down(_command_key)
+    _click_on_column_selector(canvas, 3)
+    _click_on_column_selector(canvas, 4)
+    app.keyboard.up(_command_key)
+    _click_on_row_selector(canvas, 3)
+    wait_for_app_run(app)
+
+
+def _expect_multi_row_multi_column_selection(app: Page):
+    expected = "Dataframe multi-row-multi-column selection: {'select': {'rows': [0, 2], 'columns': ['col_1', 'col_3', 'col_4']}}"
+    selection_text = app.get_by_test_id("stMarkdownContainer").filter(has_text=expected)
+    expect(selection_text).to_have_count(1)
+
+
+def test_multi_row_and_multi_column_select(app: Page):
+    canvas = _get_multi_row_and_column_select_df(app)
+    _select_some_rows_and_columns(app, canvas)
+    _expect_multi_row_multi_column_selection(app)
+
+
+def test_clear_selection_via_escape(app: Page):
+    canvas = _get_multi_row_and_column_select_df(app)
+    _select_some_rows_and_columns(app, canvas)
+
+    # make sure we have something selected before clearing it to avoid false-positives
+    _expect_multi_row_multi_column_selection(app)
+
+    app.keyboard.press("Escape")
+    wait_for_app_run(app)
+    expected = "Dataframe multi-row-multi-column selection: {'select': {'rows': [], 'columns': []}}"
+    selection_text = app.get_by_test_id("stMarkdownContainer").filter(has_text=expected)
+    expect(selection_text).to_have_count(1)
+
+
+def test_clear_selection_via_toolbar(app: Page):
+    canvas = _get_multi_row_and_column_select_df(app)
+
+    # toolbar has three buttons: download, search, fullscreen
+    data_editor_toolbar = canvas.get_by_test_id("stElementToolbar")
+    toolbar_buttons = data_editor_toolbar.get_by_test_id("stElementToolbarButton")
+    expect(toolbar_buttons).to_have_count(3)
+
+    _select_some_rows_and_columns(app, canvas)
+    _expect_multi_row_multi_column_selection(app)
+    # toolbar has one more button now: clear selection
+    toolbar_buttons = data_editor_toolbar.get_by_test_id("stElementToolbarButton")
+    expect(toolbar_buttons).to_have_count(4)
+    # click on the clear-selection button which is the first in the toolbar
+    toolbar_buttons.nth(0).click()
+    wait_for_app_run(app)
+    expected = "Dataframe multi-row-multi-column selection: {'select': {'rows': [], 'columns': []}}"
+    selection_text = app.get_by_test_id("stMarkdownContainer").filter(has_text=expected)
+    expect(selection_text).to_have_count(1)
+
+
+# def test_in_form_selection_and_session_state(app: Page):
```

Comment:
@raethlein can this test case be removed?

Reply:
yep! the same tests exists below it but in a fixed variant; this was a non-working state. So the commented out test can be removed